v4/newrelic/sql_driver_test.go

```diff
@@ -0,0 +1,295 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package newrelic
+
+import (
+	"context"
+	"database/sql/driver"
+	"strings"
+	"testing"
+
+	"github.com/newrelic/go-agent/v4/internal"
+)
+
+var (
+	driverTestMetrics = []internal.WantMetric{
+		{Name: "OtherTransaction/all", Scope: "", Forced: true, Data: nil},
+		{Name: "OtherTransaction/Go/hello", Scope: "", Forced: true, Data: nil},
+		{Name: "OtherTransactionTotalTime", Scope: "", Forced: true, Data: nil},
+		{Name: "OtherTransactionTotalTime/Go/hello", Scope: "", Forced: false, Data: nil},
+		{Name: "Datastore/all", Scope: "", Forced: true, Data: nil},
+		{Name: "Datastore/allOther", Scope: "", Forced: true, Data: nil},
+		{Name: "Datastore/mysql/all", Scope: "", Forced: true, Data: nil},
+		{Name: "Datastore/mysql/allOther", Scope: "", Forced: true, Data: nil},
+		{Name: "Datastore/operation/mysql/myoperation", Scope: "", Forced: false, Data: nil},
+		{Name: "Datastore/statement/mysql/mycollection/myoperation", Scope: "", Forced: false, Data: nil},
+		{Name: "Datastore/statement/mysql/mycollection/myoperation", Scope: "OtherTransaction/Go/hello", Forced: false, Data: nil},
+		{Name: "Datastore/instance/mysql/myhost/myport", Scope: "", Forced: false, Data: nil},
+	}
+	driverTestSpans = []internal.WantSpan{
```

Comment:
This file was just copied from the v3 directory. In addition, I added testing for spans as well, since this file in the v3 directory only tested metrics.